clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -83,7 +83,7 @@ getCategoryFromDiagGroup(const Record *Group,
 static std::string getDiagnosticCategory(const Record *R,
                                          DiagGroupParentMap &DiagGroupParents) {
   // If the diagnostic is in a group, and that group has a category, use it.
-  if (DefInit *Group = dyn_cast<DefInit>(R->getValueInit("Group"))) {
+  if (const DefInit *Group = dyn_cast<DefInit>(R->getValueInit("Group"))) {
```

Comment:
most of these are actually supposed to be `const auto *`

Reply:
Right, because of the dyn_cast<>. Let me change them